lib/src/main/java/com/auth0/jwt/oicmsg/KeyBundle.java

```diff
@@ -0,0 +1,519 @@
+            }
+
+            if (!this.remote) {
+                if (this.fileFormat.equals("jwks") || this.fileFormat.equals("jwk")) {
+                    try {
+                        this.doLocalJwk(this.source);
+                    } catch (UpdateFailed updateFailed) {
+                        logger.error("Local key updated from " + this.source + " failed.");
+                    }
+                } else if (this.fileFormat.equals("der")) {
+                    doLocalDer(this.source, this.keyType, this.keyUsage);
+                }
+            }
+        }
+    }
+
+    public String getSource() {
+        return source;
+    }
+
+    public KeyBundle() throws ImportException {
+        this(null, "", 300, true, "jwk", "RSA", null);
+    }
+
+    public KeyBundle(List<Key> keyList, String keyType) throws ImportException {
+        this(keyList, "", 300, true, "jwk", keyType, null);
+    }
+
+    public KeyBundle(List<Key> keyList, String keyType, List<String> usage) throws ImportException {
+        this(keyList, "", 300, true, "jwk", keyType, usage);
+    }
+
+    public KeyBundle(String source, boolean verifySSL) throws ImportException {
+        this(null, source, 300, verifySSL, "jwk", "RSA", null);
+    }
+
+    public KeyBundle(String source, String fileFormat, List<String> usage) throws ImportException {
+        this(null, source, 300, true, fileFormat, "RSA", usage);
+    }
+
+    public void doKeys(List<Key> keys) {
+        for (Key keyIndex : keys) {
+            final String kty = keyIndex.getKty();
+            List<String> usage = harmonizeUsage(Arrays.asList(keyIndex.getUse()));
+            keys.remove("use");
+            boolean flag = false;
+            for (String use : usage) {
+                List<String> types = new ArrayList<String>() {{
+                    add(kty);
+                    add(kty.toLowerCase());
+                    add(kty.toUpperCase());
+                }};
+                boolean isSuccess = true;
+                Key key;
+                for (String typeIndex : types) {
+                    try {
+                        switch(typeIndex) {
+                            case "RSA":
+                                key = new RSAKey("use");
+                                break;
+                            case "EC":
+                                key = new ECKey("use");
+                                break;
+                            case "SYMKey":
+                                key = new SYMKey("use");
+                                break;
+                            default:
+                                throw new IllegalArgumentException("Encryption type: " + typeIndex + " isn't supported");
+                        }
+                    } catch (JWKException exception) {
+                        logger.warn("While loading keys: " + exception);
+                        isSuccess = false;
+                    }
+
+                    if (isSuccess) {
```

Comment:
This happens if the `catch` was never called, right? Then instead of using this flag, put the code below the switch inside the `try`.

Reply:
done